### PR TITLE
feat(biome_html_analyze): implement useAnchorContent rule for HTML

### PR DESCRIPTION
**Pull request by rahuld109**
> [!NOTE]
> **AI Assistance Disclosure:** The Rust implementation in this PR was developed with Claude Code. While I have experience with web accessibility and a11y standards, I am still learning Rust.

## Summary

Implements the `useAnchorContent` accessibility rule for HTML, as part of #8155.

- Enforces that anchor elements have accessible content for screen readers
- Flags empty anchors, whitespace-only content, self-closing anchors without accessible names, and content hidden with `aria-hidden`
- Accepts `aria-label` and `title` attributes as valid accessible names (matching eslint-jsx-a11y behavior)
- Provides an unsafe fix to remove `aria-hidden` attribute when present on the anchor element

## Test plan

- [x] Added test cases for invalid HTML anchors (empty, whitespace, aria-hidden, nested hidden content, empty aria-label/title)
- [x] Added test cases for valid HTML anchors (text content, aria-label, title, nested visible content)
- [x] Added framework-specific tests for Vue, Svelte, and Astro
- [x] Verified custom components (PascalCase) don't trigger the rule
- [x] All tests pass with `cargo test -p biome_html_analyze use_anchor_content`
- [x] Clippy passes with no warnings

### CODE REVIEW

File: crates/biome_html_analyze/src/lint/a11y/use_anchor_content.rs [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        // Check if element is an anchor tag (case-insensitive for HTML)
+        let element_name = node.name()?;
+        if !element_name.text().eq_ignore_ascii_case("a") {
+            return None;
+        }
```
Comment (ematipico) [inline crates/biome_html_analyze/src/lint/a11y/use_anchor_content.rs:111]: I already reviewed this code here, and it's incorrect, did you forget? 

https://github.com/biomejs/biome/pull/8742#discussion_r2680945515

Did you use AI for these PRs?

Comment (rahuld109) [reply crates/biome_html_analyze/src/lint/a11y/use_anchor_content.rs:111]: Yes, updated the PR description to specify the same.

File: crates/biome_html_analyze/src/lint/a11y/use_anchor_content.rs [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+                    .is_some_and(|attribute| {
+                        attribute
+                            .initializer()
+                            .and_then(|init| init.value().ok())
+                            .and_then(|value| value.string_value())
+                            .is_none_or(|value| !value.eq_ignore_ascii_case("false"))
+                    });
+
+            !has_aria_hidden
+        }
```
[the remark this answers is not in the document]
Comment (ematipico) [reply crates/biome_html_analyze/src/lint/a11y/use_anchor_content.rs:330]: @coderabbitai I believe you misunderstood the implementation of the rule. The rule is applied **to HTML**, not JSX.  You also know that self-closing elements don't exist in HTML. Knowing this, re-run your heuristics, and if the suggestion is still applicable, reframe it in an HTML context.

File: crates/biome_html_analyze/src/lint/a11y/use_anchor_content.rs [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+        AnyHtmlElement::HtmlBogusElement(_) | AnyHtmlElement::HtmlCdataSection(_) => true,
+    })
+}
+
+/// Checks if the content node contains non-empty text.
+fn is_accessible_text_content(content: &AnyHtmlContent) -> bool {
+    match content {
+        AnyHtmlContent::HtmlContent(html_content) => html_content
+            .value_token()
+            .is_ok_and(|token| !token.text_trimmed().trim().is_empty()),
```
Comment (ematipico) [inline crates/biome_html_analyze/src/lint/a11y/use_anchor_content.rs]:
```suggestion
            .is_ok_and(|token| !token.text_trimmed().is_empty()),
```

no need to trim twice

**Review by ematipico — APPROVED** [review]

File: crates/biome_html_analyze/src/lint/a11y/use_anchor_content.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ematipico) [reply crates/biome_html_analyze/src/lint/a11y/use_anchor_content.rs:330]: @rahuld109, you might want to look at this comment

[the remark this answers is not in the document]
Comment (rahuld109) [reply crates/biome_html_analyze/src/lint/a11y/use_anchor_content.rs:330]: aded the negative test cases along with the handling of this edge cases.
